CIP-XXXX/README.md

```diff
@@ -0,0 +1,205 @@
+transparently in the Plutus library by adding another version of evaluateScriptRestricting
+with an equivalent type but for a new version of Plutus.
+
+## Rationale
+
+Plutus isn't made lazy by this change, errors are instead not immediately propagated up.
+Errors are less special-cased now, as evident by the removal of a rule, not addition.
+
+One (somewhat obvious) issue is that `error` isn't the only source of divergence.
+As @effectfully pointed out, the following is still problematic:
+```haskell
+let factorial n = ifThenElse (n == 0) 1 (factorial (n - 1))
+in factorial 2
+```
+The arguments to `ifThenElse` are evaluated first, leading to infinite recursion.
+Fixing this would require making Plutus to some extent lazy, which is a much larger change.
+
+Is this change still worth it in light of this?
+Even _purely_ for the sake of making scripts more functional, it would still be worth it.
+Very few people will make their own `loopforever` function to replicate the old `error` to write impure scripts.
+It is a bonus that we in addition 1) simplify the language 2) remove a foot-gun for new Plutusers/Plutors/Plutonites
+3) make Plutus scripts more efficient by removing `delay` and `force` from the erasure.
+
+It is not immediately obvious that the third step is possible, as we still have the above issue.
+The common example (from [Well-Typed](https://well-typed.com/blog/2022/08/plutus-cores/)) is:
+```haskell
+let* !boom = λ {a} -> ERROR
+     !id   = λ {a} (x :: a) -> x
+in λ (b :: Bool) -> ifThenElse {∀ a. a -> a} b boom id
+```
+Indeed, this example works _without_ adding delays to type abstractions,
+as the untyped form
+```haskell
+f = λ b -> ifThenElse b ERROR (λ x -> x)
+```
+is precisely the example we seek to make work.
+`f True ()` will fail as expected,
+while `f False ()` will return `()` as expected.
+
+### Soundness of optimisations
+
+We note that you can soundly optimise `(\x -> M) N` into `M` if `x` is not mentioned in `M`,
+or similar occurrences where the result of `N` isn't needed.
+Such an optimisation will change the behaviour of the script necessarily,
+as any traces in `N` will no longer happen, yet, this is not important in practice
+as traces are merely a debugging tool. The budget used will decrease,
+but people do not rely on budget exhaustion to signal invalidity.
+
+Transformations like `M || N` into `force (ifThenElse M (delay True) (delay N))` follow a similar principle:
+traces that happened before might no longer happen.
+The used budget will increase if M is `False`, yet this is likely
+still an acceptable transformation for most people.
+
+### Backward compatibility
+
+This CIP has the issue of old scripts not being portable to the new version trivially.
+You need to make sure your script isn't making use of any impure assumptions.
+
+## Path to Active
+
+### Acceptance Criteria
+
+It would be needed to implement the necessary changes in `cardano-ledger` and `plutus`.
+
+### Implementation Plan
```

Comment:
```suggestion
### Acceptance Criteria

It would be needed to implement the necessary changes in `cardano-ledger` and `plutus`.

### Implementation Plan
```
We need more detail about how the both the Ledger & Plutus teams would be expected to verify this.  I was hoping to use the similarly split https://github.com/cardano-foundation/CIPs/pull/758 as an example, but that too has been stalled & is incomplete in that regard.

@zliu41 @lehins can you please provide some help about what exactly would be needed here for verification of this proposal?

Reply:
I've commented down below, I think this proposal can't work as written, so it can't be implemented and verified either. I may be wrong of course, so let's see what further discussion shows.